### PR TITLE
Lordaeron changes

### PR DESCRIPTION
**Pull request by SebastianTanoni**

### CODE REVIEW

File: src/WarcraftLegacies.Source/Quests/Lordaeron/QuestCapitalCity.cs [hunk 1]
```diff
@@ -28,13 +30,13 @@ public sealed class QuestCapitalCity : QuestData
...
     /// </summary>
-    public QuestCapitalCity(Rectangle rescueRect, unit terenas, LegendaryHero uther,
+    public QuestCapitalCity(PreplacedUnitSystem preplacedUnitSystem, Rectangle rescueRect, unit terenas, LegendaryHero uther,
       LegendaryHero arthas, Capital caerDarrow, Capital capitalPalace, IEnumerable<QuestData> prequisites) :
       base("Hearthlands",
         "The territories of Lordaeron are fragmented. Regain control of the old Alliance's hold to secure the kingdom.",
         @"ReplaceableTextures\CommandButtons\BTNCastle.blp")
     {
-      AddObjective(new ObjectiveControlCapital(caerDarrow, false));
+      AddObjective(new ObjectiveUnitIsDead(preplacedUnitSystem.GetUnit(FourCC("n0AG"))));
```
Comment (YakaryBovine) [inline src/WarcraftLegacies.Source/Quests/Lordaeron/QuestCapitalCity.cs:39]: Is the `Constants` version of this not showing up?

File: src/WarcraftLegacies.Source/Quests/Lordaeron/QuestChampionOfTheLight.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      ResearchId = Constants.UPGRADE_R01Q_QUEST_COMPLETED_CHAMPION_OF_THE_LIGHT;
+    }
+
+    /// <inheritdoc />
+    protected override string RewardFlavour =>
+      "Uther has achieved the status of living legend, inspiring the men and women of Lordaeron to strive for greatness.";
+
+    /// <inheritdoc />
+    protected override string RewardDescription =>
+      $"Your casters and paladin gain 200 hit points and 5 damage. Paladin gain the reincarnation ability.";
```
Comment (YakaryBovine) [inline src/WarcraftLegacies.Source/Quests/Lordaeron/QuestChampionOfTheLight.cs:29]:
```suggestion
      $"Your casters and Paladins gain 200 hit points and 5 damage, and Paladins gain the Reincarnation ability";
```

File: src/WarcraftLegacies.Source/Quests/Lordaeron/QuestKingArthas.cs [hunk 3]
```diff
@@ -38,11 +42,11 @@ public sealed class QuestKingArthas : QuestData
 
     /// <inheritdoc/>
     protected override string RewardFlavour =>
-      "With the Lich King eliminated, the Kingdom of Lordaeron is free of its greatest threat. King Terenas Menethil proudly abdicates in favor of his son.";
+      "With a buffer outpost in Northrend, the Kingdom of Lordaeron is safer of its greatest threat. King Terenas Menethil proudly abdicates in favor of his son, now an experienced warrior.";
 
     /// <inheritdoc/>
     protected override string RewardDescription =>
-      "Arthas becomes the King of Lordaeron, gains the Crown of Lordaeron, and he can no longer permanently die";
+      $"Arthas becomes the King of Lordaeron, gains the Crown of Lordaeron, and he can no longer permanently die. You can now build {GetObjectName(Constants.UNIT_H06C_HIGH_TOWER_LORDAERON_SPECIALIST)}. Your Knights and Huntsman will also be upgraded to stronger units.";
```
Comment (YakaryBovine) [inline src/WarcraftLegacies.Source/Quests/Lordaeron/QuestKingArthas.cs:49]:
```suggestion
      $"Arthas becomes the King of Lordaeron, gains the Crown of Lordaeron, and he can no longer permanently die. Learn to build {GetObjectName(Constants.UNIT_H06C_HIGH_TOWER_LORDAERON_SPECIALIST)}s. Your Knights and Huntsman are upgraded to stronger units";
```
I think you also want to say like, "Knights and Humans are upgraded to x and y". This is a bit vague right now.

Comment (YakaryBovine) [inline src/WarcraftLegacies.Source/Quests/Lordaeron/QuestKingArthas.cs:49]: Use `GetObjectName` for the unit names.

Comment (SebastianTanoni) [reply src/WarcraftLegacies.Source/Quests/Lordaeron/QuestKingArthas.cs:49]: i was afraid the text would be too long

File: src/WarcraftLegacies.Source/Quests/Lordaeron/QuestKingArthas.cs [hunk 4]
```diff
@@ -38,11 +42,11 @@ public sealed class QuestKingArthas : QuestData
 
     /// <inheritdoc/>
     protected override string RewardFlavour =>
-      "With the Lich King eliminated, the Kingdom of Lordaeron is free of its greatest threat. King Terenas Menethil proudly abdicates in favor of his son.";
+      "With a buffer outpost in Northrend, the Kingdom of Lordaeron is safer of its greatest threat. King Terenas Menethil proudly abdicates in favor of his son, now an experienced warrior.";
```
Comment (YakaryBovine) [inline src/WarcraftLegacies.Source/Quests/Lordaeron/QuestKingArthas.cs:45]: This breaks the flavour of the quest. The description says that the only thing standing in the way of Arthas' coronation is the Scourge, but then you don't actually have to deal with the Scourge to complete it. Possibly needs a total rewrite.

Comment (SebastianTanoni) [reply src/WarcraftLegacies.Source/Quests/Lordaeron/QuestKingArthas.cs:45]: Allright, il try and think of something

File: src/WarcraftLegacies.Source/Quests/Lordaeron/QuestTyrHand.cs [hunk 5]
```diff
@@ -29,19 +30,19 @@ public sealed class QuestTyrHand : QuestData
...
+      ResearchId = Constants.UPGRADE_R023_QUEST_COMPLETED_THE_FORTIFIED_CITY;
       Required = true;
     }
 
     /// <inheritdoc />
     protected override string RewardFlavour => "The city-fortress of Tyr's Hand has decided to join us!";
 
     /// <inheritdoc />
-    protected override string RewardDescription => "Control of all units in Tyr's Hand and Garithos is trainable";
+    protected override string RewardDescription => $"Control of all units in Tyr's Hand and Garithos is trainable. You can also build the {GetObjectName(Constants.UNIT_H094_SIEGE_WORKSHOP_LORDAERON_SIEGE)}.";
```
Comment (YakaryBovine) [inline src/WarcraftLegacies.Source/Quests/Lordaeron/QuestTyrHand.cs:45]:
```suggestion
    protected override string RewardDescription => $"Gain control of all units in Tyr's Hand, learn to train Garithos from the x, and learn to build {GetObjectName(Constants.UNIT_H094_SIEGE_WORKSHOP_LORDAERON_SIEGE)}s";
```

File: src/WarcraftLegacies.Source/Quests/Lordaeron/QuestTyrHand.cs [hunk 6]
```diff
@@ -29,19 +30,19 @@ public sealed class QuestTyrHand : QuestData
...
+      AddObjective(new ObjectiveEitherOf(new ObjectiveCapitalDead(stratholme),
+        new ObjectiveTime(840)));
       AddObjective(new ObjectiveSelfExists());
       _rescueUnits = rescueRect.PrepareUnitsForRescue(RescuePreparationMode.HideNonStructures);
+      ResearchId = Constants.UPGRADE_R023_QUEST_COMPLETED_THE_FORTIFIED_CITY;
       Required = true;
     }
 
     /// <inheritdoc />
     protected override string RewardFlavour => "The city-fortress of Tyr's Hand has decided to join us!";
```
Comment (YakaryBovine) [inline src/WarcraftLegacies.Source/Quests/Lordaeron/QuestTyrHand.cs:42]: From a flavor standpoint - need to think about why Tyr's Hand gives you siege weapons. I think it'd be fine making something up here about them having an abundance of workshops or something, I don't know.

File: src/WarcraftLegacies.Source/Quests/Scourge/QuestScouringofLordaeron.cs [hunk 7]
```diff
@@ -11,18 +11,18 @@ public sealed class QuestScouringofLordaeron : QuestData
...
-      foreach (var capital in capitalTargets)
-        AddObjective(new ObjectiveCapitalDead(capital));
+      AddObjective(new ObjectiveControlCapital(capitalPalace, false));
+      AddObjective(new ObjectiveCapitalDead(stratholme));
     }
 
     /// <inheritdoc/>
     protected override string RewardFlavour =>
-      "The capital of Lordaeron has fallen, weakening Lordaeron's Champion.";
+      "The capitals of Lordaeron has fallen, weakening Lordaeron's Champion.";
```
Comment (YakaryBovine) [inline src/WarcraftLegacies.Source/Quests/Scourge/QuestScouringofLordaeron.cs:25]: Lorewise I don't think Stratholme is a capital. Maybe like, the "major cities of Lordaeron have fallen"?
Also it's an old problem but some explanation of why this weakens Uther would be a huge improvement.

File: src/WarcraftLegacies.Source/Setup/Legends/LegendLordaeron.cs [hunk 8]
```diff
@@ -66,6 +66,7 @@ public LegendLordaeron(PreplacedUnitSystem preplacedUnitSystem, ArtifactSetup ar
       CapitalPalace = new Capital
       {
         Unit = preplacedUnitSystem.GetUnit(Constants.UNIT_H000_CAPITAL_PALACE_LORDAERON),
+        Capturable = true,
         DeathMessage = "The capital city of Lordaeron has been razed, and King Terenas is dead.",
```
Comment (YakaryBovine) [inline src/WarcraftLegacies.Source/Setup/Legends/LegendLordaeron.cs:70]: No point having a death message on a Capturable base.

File: src/WarcraftLegacies.Source/Quests/Scourge/QuestScouringofLordaeron.cs (hunk 7 above)
Comment (SebastianTanoni) [reply src/WarcraftLegacies.Source/Quests/Scourge/QuestScouringofLordaeron.cs:25]: Major cities will sound better!

Comment (SebastianTanoni) [reply src/WarcraftLegacies.Source/Quests/Scourge/QuestScouringofLordaeron.cs:25]: Could remove the weakning part to be honest. and just remove the quest, and make him unrevivable when the 2 bases are done

File: src/WarcraftLegacies.Source/Quests/Lordaeron/QuestKingArthas.cs (hunk 3 above)
Comment (YakaryBovine) [reply src/WarcraftLegacies.Source/Quests/Lordaeron/QuestKingArthas.cs:49]: That window has a scroll bar so it's not really a concern. If the player doesn't scroll, they get the same information they do now. If they do, they get all the info.

Comment (SebastianTanoni) [reply src/WarcraftLegacies.Source/Quests/Lordaeron/QuestKingArthas.cs:49]: ok!

**Review by YakaryBovine — CHANGES_REQUESTED** [review]
Forgot to press the "Request changes" button so doing it now!

File: src/WarcraftLegacies.Source/Quests/Lordaeron/QuestKingArthas.cs (hunk 4 above)
Comment (SebastianTanoni) [reply src/WarcraftLegacies.Source/Quests/Lordaeron/QuestKingArthas.cs:45]: i tried something, you can probably word it better

File: src/WarcraftLegacies.Source/Quests/Lordaeron/QuestKingArthas.cs [hunk 9]
```diff
@@ -21,13 +23,15 @@ public sealed class QuestKingArthas : QuestData
     /// <summary>
     /// Initializes a new instance of the <see cref="QuestKingArthas"/> class.
     /// </summary>
-    public QuestKingArthas(unit terenas, Artifact crownOfLordaeron, Capital capitalPalace, LegendaryHero arthas, Capital lichKing) : base("Line of Succession",
-      "Arthas Menethil is the one true heir of the Kingdom of Lordaeron. The only thing standing in the way of his coronation is the world-ending threat of the Scourge.",
+    public QuestKingArthas(unit terenas, Artifact crownOfLordaeron, Capital capitalPalace, LegendaryHero arthas) : base("Line of Succession",
+      "Arthas Menethil is the one true heir of the Kingdom of Lordaeron. The only thing standing in the way of his coronation is the world-ending threat of the Scourge. Lordaeron is vulnerable, Arthas will need to safeguard his kingdom and prove his worth to ascend as king.",
```
Comment (YakaryBovine) [inline src/WarcraftLegacies.Source/Quests/Lordaeron/QuestKingArthas.cs:27]:
```suggestion
      "Arthas Menethil is the sole heir to the Lordaeron crown. His father, ever obstinate in his old age, denies the existential threat of the Scourge and forbids Arthas from bringing the fight to Northrend. The crown prince will simply have to take matters into his own hands.",
```

File: src/WarcraftLegacies.Source/Quests/Lordaeron/QuestKingArthas.cs [hunk 10]
```diff
@@ -38,11 +42,13 @@ public sealed class QuestKingArthas : QuestData
 
     /// <inheritdoc/>
     protected override string RewardFlavour =>
-      "With the Lich King eliminated, the Kingdom of Lordaeron is free of its greatest threat. King Terenas Menethil proudly abdicates in favor of his son.";
+      "With a succesful offensive in Northrend and the establishement of an outpost in the Holwing Fjord, the Kingdom of Lordaeron is safer of its greatest threat. King Terenas Menethil proudly abdicates in favor of his son, now an experienced warrior.";
```
Comment (YakaryBovine) [inline src/WarcraftLegacies.Source/Quests/Lordaeron/QuestKingArthas.cs:45]:
```suggestion
      "Fate decreed that Arthas would witness the fall of Stratholme and become corrupted by vengeance. Instead, he defended his homeland from the ravenous Scourge and took the battle to Northrend. Back at home, Terenas Menethil is forced to admit: his son is ready to be King.";
```

**Comment by YakaryBovine** [comment]
image
Can we just get rid of the "Arthas Menethil is at King Terenas" bit? Seems a bit silly considering it would require Arthas to go back home before the Frozen Throne is destroyed.

**Comment by YakaryBovine** [comment]
image
This new ability shares an icon with Transmute.

**Comment by YakaryBovine** [comment]
image
This quest says that Arthas can "no longer permanently die", but actually it doesn't affect his permadeath status at all.

**Review by YakaryBovine — CHANGES_REQUESTED** [review]
Some assorted issues to fix. The Arthas stuff might be complicated, let me know if you need help.

**Comment by Ripzss** [comment]
> image Can we just get rid of the "Arthas Menethil is at King Terenas" bit? Seems a bit silly considering it would require Arthas to go back home before the Frozen Throne is destroyed.

Couldnt come up with any more fitting Icons if u can think of any i can change it

**Comment by Ripzss** [comment]
> image This new ability shares an icon with Transmute.

I couldnt come up with any other icons if u can think of any more fitting i can change it.

**Comment by YakaryBovine** [comment]
> > image This new ability shares an icon with Transmute.
> 
> 
> 
> I couldnt come up with any other icons if u can think of any more fitting i can change it. 

As the person adding the ability it's up to you to find and implement the icon. This is a very generic ability so just look up Holy on Hive Workshop and you'll find something.

**Review by YakaryBovine — APPROVED** [review]

**Pull request merged**